docs/data/date-pickers/validation/validation.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Date and Time pickers - Validation
+---
+
+# Date and Time pickers - Validation
+
+<p class="description">Add custom validation to user inputs.</p>
+
+All the date and time pickers have an API for adding validation constraints.
+By default they provide visual feedback if the component value doesn't meet the validation criteria.
+
+:::info
+The validation props are showcased for each type of picker component (`TimePicker`, `DatePicker`, `DateRangePicker`, etc).
+But the same props are available on all component variants.
+
+For example—the validation props available on the `DatePicker` component are also available on:
+
+- `DesktopDatePicker`
+- `MobileDatePicker`
+- `StaticDatePicker`
+
+:::
+
+## Invalid values feedback
+
+On the input—it enables its error state.
+
+{{"demo": "ValidationBehaviorInput.js", "defaultCodeOpen": false}}
+
+On the calendar and clock views—the invalid values are displayed as disabled to prevent their selection.
+
+{{"demo": "ValidationBehaviorView.js", "defaultCodeOpen": false}}
+
+## Past and future validation
+
+All pickers support the past and future validation.
+
+The `disablePast` prop prevents the selection of all values before today.
+
+- On the `day` view—all the days before today won't be selectable.
+- On the `month` and `year` views—all the values ending before today won't be selectable.
+
+{{"demo": "DateValidationDisablePast.js", "defaultCodeOpen": false}}
+
+The `disableFuture` prop prevents the selection of all values after today.
+
+- On the `day` view—all the days after today won't be selectable.
+- On the `month` and `year` views—all the values beginning after today won't be selectable.
+
+{{"demo": "DateValidationDisableFuture.js", "defaultCodeOpen": false}}
+
+## Date validation
+
+All the props described below are available on all the components supporting date edition.
+
+### Set date boundaries
+
+The `minDate` prop prevents the selection of all values before `props.minDate`.
+
+- On the `day` view—all the days before the `minDate` won't be selectable.
+- On the `month` and `year` views—all the values ending before the `minDate` won't be selectable.
```

Comment:
A few points:
- we are talking about `year, month, day` views, but by default date pickers don't include `month` view and that won't be seen in the demo, but is mentioned. I'd say we should add explicit views for these demos, like in this case `views={['year', 'month', 'day']}`
- demos include a `DateTimePicker`, but text does not mention `hour, minutes, seconds` views. Should we?

Reply:
I added the views and some details for the time views

Reply:
I was thinking about something along these lines: add every possible view to each of the component.
What do you think? 🤔 
Also, in the text we mention `seconds` view, but it is nowhere to be seen, maybe including it in the views would make more sense here? 🤔 
But again, if we add `seconds` view it might make sense to have an option to reset 'now'.
What do you think about the following diff?
```diff
diff --git a/docs/data/date-pickers/validation/DateValidationDisablePast.tsx b/docs/data/date-pickers/validation/DateValidationDisablePast.tsx
index cc0db38c6..b373671b6 100644
--- a/docs/data/date-pickers/validation/DateValidationDisablePast.tsx
+++ b/docs/data/date-pickers/validation/DateValidationDisablePast.tsx
@@ -10,6 +10,7 @@ import { DatePicker } from '@mui/x-date-pickers/DatePicker';
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
 import { DateRange, DateRangePicker } from '@mui/x-date-pickers-pro/DateRangePicker';
 import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+import Button from '@mui/material/Button';
 
 const today = dayjs();
 const yesterday = dayjs().subtract(1, 'day');
@@ -47,10 +48,12 @@ export default function DateValidationDisablePast() {
   const [dateRangePickerValue, setDateRangePickerValue] = React.useState<
     DateRange<Dayjs>
   >([yesterday, today]);
+  const [key, setKey] = React.useState(dayjs().millisecond());
 
   return (
     <LocalizationProvider dateAdapter={AdapterDayjs}>
-      <Stack spacing={4}>
+      <Stack spacing={4} key={key}>
+        <Button onClick={() => setKey(dayjs().millisecond())}>Re-render</Button>
         <GridItem label="DatePicker">
           <DatePicker
             disablePast
@@ -66,6 +69,7 @@ export default function DateValidationDisablePast() {
             value={dateTimePickerValue}
             onChange={(newValue) => setDateTimePickerValue(newValue)}
             renderInput={(params) => <TextField {...params} />}
+            views={['year', 'month', 'day', 'hours', 'minutes', 'seconds']}
           />
         </GridItem>
         <GridItem label="TimePicker">
@@ -74,6 +78,7 @@ export default function DateValidationDisablePast() {
             value={timePickerValue}
             onChange={(newValue) => setTimePickerValue(newValue)}
             renderInput={(params) => <TextField {...params} />}
+            views={['hours', 'minutes', 'seconds']}
           />
         </GridItem>
         <GridItem label="DateRangePicker" spacing={2}>
```

Reply:
> I was thinking about something along these lines: add every possible view to each of the component.
What do you think? 

OK, I was lazy but you are right

Reply:
For the seconds, how about we just don't talk about the seconds since it's a very weird use case ?
With the `reset` button I feel it impacts a lot the whole demo for an edge case.

Or maybe add a small demo inside my `info` block with only a `TimePicker` with only the `seconds` view and we log the current second in a div about the picker.

Reply:
I added the `month` view to the maxDate / minDate / shouldDisableDate demos as well

Reply:
> For the seconds, how about we just don't talk about the seconds since it's a very weird use case ?
With the reset button I feel it impacts a lot the whole demo for an edge case.
Or maybe add a small demo inside my info block with only a TimePicker with only the seconds view and we log the current second in a div about the picker.

Yeah, sure, that was just a suggestion, but I do agree, that these are possibly quite a niche edge-cases that might not deserve this much attention. 👌